### PR TITLE
Export region countries data

### PR DESCRIPTION
**Pull request by nickoola**

### CODE REVIEW

File: src/phoneField/index.tsx [hunk 1]
```diff
@@ -120,16 +122,19 @@ export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFiel
     countriesWhiteList: this.props.countriesWhiteList,
     [FIELDS.PHONENUMBER]: this.props.defaultValue,
     [FIELDS.PHONEPREFIX]: iso2toDialCode(this.props.defaultPrefix),
+    phoneRegion: [this.props.defaultPrefix],
```
Comment (grosbouddha) [inline src/phoneField/index.tsx]: Rename phoneRegion to phoneRegionList

File: src/phoneField/index.tsx [hunk 2]
```diff
@@ -120,16 +122,19 @@ export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFiel
...
 
   onChange = ({ name, value }: OnChangeParameters) => {
     this.setState({ [name]: value }, () => {
+      const iso2Codes = allCountriesCode[Number(this.state[FIELDS.PHONEPREFIX])]
       this.props.onChange({
         name: this.props.name,
         value: {
           [FIELDS.PHONENUMBER]: this.state[FIELDS.PHONENUMBER],
           [FIELDS.PHONEPREFIX]: this.state[FIELDS.PHONEPREFIX],
+          phoneRegion: !isEmpty(iso2Codes) ? iso2Codes : [''],
```
Comment (grosbouddha) [inline src/phoneField/index.tsx]: Can not we use an empty array instead of this empty magic string?

File: src/phoneField/index.tsx [hunk 3]
```diff
@@ -33,27 +35,27 @@ interface PhoneFieldProps {
...
   countriesData: mappedCountriesPhoneData[]
   countriesWhiteList: string[]
+  phoneRegion: string[]
   completePhoneNumber: string
   // Make the type of [key: string]: string etc... assignable for setState()
   [key: string]: any
 }
 
 const allCountriesPhoneData: allCountriesPhoneData[] = countryTelData.allCountries
+const allCountriesCode: allCountriesCode = countryTelData.allCountryCodes
```
Comment (loick) [inline src/phoneField/index.tsx]: Can't you destructure countryTelData directly from the import?

File: src/phoneField/index.tsx [hunk 4]
```diff
@@ -98,7 +100,7 @@ const iso2toDialCode = (iso2Value: string) => {
   }
   const countryPhoneData = findIso2(iso2Value)
   if (countryPhoneData) {
-    return `+${countryPhoneData.dialCode}`
+    return countryPhoneData.dialCode
```
Comment (loick) [inline src/phoneField/index.tsx]: What is this update related to? It is not reflected in the changelog

File: src/phoneField/index.tsx (hunk 1 above)
Comment (loick) [inline src/phoneField/index.tsx]: if you don't find the phoneRegion, maybe you should add a default value ([])

File: src/phoneField/index.tsx (hunk 2 above)
Comment (loick) [inline src/phoneField/index.tsx]: why is there an empty string here?

File: src/phoneField/index.unit.tsx [hunk 5]
```diff
@@ -56,6 +56,7 @@ describe('PhoneField', () => {
       countriesWhiteList,
       phoneNumber: '',
       phonePrefix: '',
+      phoneRegion: [''],
```
Comment (loick) [inline src/phoneField/index.unit.tsx]: Same here

File: src/phoneField/index.tsx [hunk 6]
```diff
@@ -75,10 +77,10 @@ const mapCountriesPhoneData = (
   countriesName: customCountriesName,
 ): mappedCountriesPhoneData[] => {
   return countriesData.map(countryData => ({
-    value: `+${countryData.dialCode}`,
+    value: countryData.dialCode,
```
Comment (grosbouddha) [inline src/phoneField/index.tsx]: Could we rename value (which has not intrinsic meaning) to something more semantic to help readability? maybe dialCode or isdCode ?

File: src/phoneField/specifications/default.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Default usage
+
+```js
+<PhoneField name="phoneField" onChange={() => {}} />
+```
+
+## Behaviour
+
+On small devices on SPA the country selector and the phone input field should get separated into two separate fields.
```
Comment (grosbouddha) [inline src/phoneField/specifications/default.md]: SPA might not mean much for external users. Not sure how much we care though.

File: src/phoneField/index.tsx (hunk 4 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: I added the "+" directly into `formattedCountriesPhoneData`, avoiding re-adding it on many place of the code.
I'll add it into the changelog.

File: src/phoneField/index.tsx (hunk 1 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: It's already the case on defaultProps >> `defaultPrefix: '',`
And into the onChange function >> `phoneRegion: !isEmpty(iso2Codes) ? iso2Codes : [''],`

File: src/phoneField/index.tsx (hunk 4 above)
Comment (wakooka) [reply src/phoneField/index.tsx]: @loick If it's an internal change, it shouldn't be in the Changelog

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,6 +6,7 @@
 
 - **[BREAKING CHANGE]** Migrate `DatePicker` to ReactDayPicker instead of React Dates, parts of the API of the component has been changed, and it no longer supports any language by default
 - **[BREAKING CHANGE]** Remove Moment dependency
+- **[UPDATE]** `PhoneField` Export countries phone data
```
[the remark this answers is not in the document]
Comment (nickoola) [reply CHANGELOG.md]: Done

File: src/phoneField/index.tsx (hunk 2 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: Done

File: src/phoneField/index.tsx (hunk 3 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: Done

File: src/phoneField/index.tsx (hunk 1 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: Done

File: src/phoneField/index.unit.tsx (hunk 5 above)
Comment (nickoola) [reply src/phoneField/index.unit.tsx]: Done

File: src/phoneField/index.tsx (hunk 6 above)
Comment (nickoola) [reply src/phoneField/index.tsx]: I had to keep the object format {value: '', label: ''} to match with the SelectField API component to build options.

File: src/phoneField/specifications/default.md (hunk 7 above)
Comment (nickoola) [reply src/phoneField/specifications/default.md]: Yeah I'l adapt a bit this content.

File: src/typings/vendors.d.ts [hunk 9]
```diff
@@ -3,6 +3,6 @@ declare module '*.md'
 declare module 'country-telephone-data' {
   const countryTelData: countryTelData
   const allCountries: allCountriesPhoneData[]
-  const allCountryCodes: allCountryCodes
+  const allCountryCodes: allCountriesCode
```
Comment (Tuxosaurus) [inline src/typings/vendors.d.ts]: why not keeping `allCountryCodes`?

File: src/phoneField/specifications/custom.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    name="phoneField"
+    className="additionalClass"
+    ariaLabelledBy="labelId"
+    selectFieldLabel="Phone country code"
+    textFieldTitle="Phone number"
+    textFieldPlaceholder="Phone number"
+    defaultRegionValue="FR"
+    defaultPhoneValue="0778674534"
+    countriesWhiteList={['FR', 'ES']}
+    customCountriesName={{
```
Comment (Tuxosaurus) [inline src/phoneField/specifications/custom.md]: nit:
countriesWhiteList => countryWhitelist
customCountriesName => customCountryNames

File: src/typings/vendors.d.ts (hunk 9 above)
Comment (nickoola) [reply src/typings/vendors.d.ts]: Yep it's finally better to keep the library naming

Comment (nickoola) [reply src/typings/vendors.d.ts]: Done

File: src/phoneField/specifications/custom.md (hunk 10 above)
Comment (nickoola) [reply src/phoneField/specifications/custom.md]: I finally do a logic renaming of const and types.
